src/trace_processor/perfetto_sql/stdlib/android/cpu/cpu_uptime.sql

```diff
@@ -0,0 +1,195 @@
+--
+-- Copyright 2024 The Android Open Source Project
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--     https://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+INCLUDE PERFETTO MODULE linux.cpu.idle;
+
+CREATE PERFETTO TABLE _sched_processes_for_span AS
+SELECT
+  ss.ts,
+  ss.dur,
+  ss.cpu,
+  ss.id,
+  t.tid,
+  t.name AS thread_name,
+  p.pid,
+  p.name AS process_name
+FROM sched_slice AS ss
+JOIN thread AS t
+  USING (utid)
+JOIN process AS p
+  USING (upid)
+WHERE
+  p.name IS NOT NULL;
+
+CREATE PERFETTO TABLE _cpu_active_for_span AS
+WITH
+  step1 AS (
+    SELECT
+      ts,
+      dur,
+      cpu,
+      ts AS group_ts,
+      dur AS group_dur,
+      lead(ts) OVER (PARTITION BY cpu ORDER BY ts) AS next_group_ts
+    FROM cpu_idle_counters
+    WHERE
+      idle = -1
+  )
+SELECT
+  *,
+  row_number() OVER (ORDER BY ts) AS group_id
+FROM step1;
+
+CREATE VIRTUAL TABLE _android_active_sched_joined USING SPAN_JOIN (_cpu_active_for_span PARTITIONED cpu, _sched_processes_for_span PARTITIONED cpu);
+
+-- Table which groups scheduling information along with the CPU idle state
+-- information. This is meant to be used in conjunciton with the android_cpu_uptime_cost
+-- calculation macro.
+CREATE PERFETTO TABLE android_active_sched_joined (
```

Comment:
Really don't like this table name. Please be more descriptive and consistent with the style of the stdlib instead of just calling it "joined"